### PR TITLE
ROU-4527 - [OSUI] - Accessibility improvements identified on team audit for the Rating

### PR DESCRIPTION
**Pull request by JoaoFerreira-FrontEnd**
This PR is for improve rating component in terms of accessibility.

### What was done

- add logic to manage aria-hidden and area checked;
- remove the logic that was adding wcaghideText to the first input label;
- improve scss file;
- add styles to prevent half from being visible when is not half rating;
- add styles to guarantee the empty editable rating is visible;

### Test Steps

1. Open the test page
2. Run an A11Y checker
3. Run the page using a screen reader
4. Expected: No warnings related to rating

### Screenshots

(prefer animated gif)

### Checklist

-   [ ] tested locally
-   [ ] documented the code
-   [ ] clean all warnings and errors of eslint
-   [ ] requires changes in OutSystems (if so, provide a module with changes)
-   [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts [hunk 1]
```diff
@@ -169,16 +176,20 @@ namespace OSFramework.OSUI.Patterns.Rating {
...
 			// if not first input, which is hidden, add the html stored form the placeholders
 			const labelHTML = index !== 0 ? this._clonedPlaceholders : '';
 			// Create a unique rating input id, based on the index
 			const ratingInputId: string = this.uniqueId + '-rating-' + index;
 
-			// Craete input and label html
-			const input = `<input type="radio" class="${Enum.CssClass.RatingInput} ${Enum.CssClass.WCAGHideText}" id=${ratingInputId} name=${this._ratingInputName} value=${index}/>`;
-			const label = `<label class='${Enum.CssClass.RatingItem} ${hideLabelClass}' for=${ratingInputId}><span class="${Enum.CssClass.WCAGHideText}">Rating ${index}</span>${labelHTML}</label>`;
+			// Create input and label html
+			const input = `<input type="radio" class="${Enum.CssClass.RatingInput} ${Enum.CssClass.WCAGHideText}" id=${ratingInputId} name=${this._ratingInputName} value=${index} aria-hidden="true">`;
```
Comment (bmarcelino-fe) [inline src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:185]: It's possible to enum the types and aria roles? If it's possible, lets avoid this on our code and change to enums

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts [hunk 2]
```diff
@@ -169,16 +176,20 @@ namespace OSFramework.OSUI.Patterns.Rating {
...
+
+			let label;
+			if (!this.configs.IsEdit) {
+				label = `<label class='${Enum.CssClass.RatingItem}' for=${ratingInputId} aria-hidden="true"><span class="${Enum.CssClass.WCAGHideText}">Rating ${index}</span>${labelHTML}</label>`;
```
Comment (bmarcelino-fe) [inline src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:189]: Same as above

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts [hunk 3]
```diff
@@ -329,6 +351,7 @@ namespace OSFramework.OSUI.Patterns.Rating {
 			// If there's only one rating item, then there's no need for further checks, this one will be checked
 			if (this.configs.RatingScale === 1) {
 				ratingItems[1].checked = true;
+				ratingItems[1].ariaChecked = 'true';
```
Comment (bmarcelino-fe) [inline src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:354]: Same as above

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts [hunk 4]
```diff
@@ -358,6 +381,7 @@ namespace OSFramework.OSUI.Patterns.Rating {
 
 			// Set the itemas as :checked
 			ratingItems[newValue].checked = true;
+			ratingItems[newValue].ariaChecked = 'true';
```
Comment (bmarcelino-fe) [inline src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:384]: Same as above

**Review by bmarcelino-fe — CHANGES_REQUESTED** [review]

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts [hunk 5]
```diff
@@ -142,6 +142,13 @@ namespace OSFramework.OSUI.Patterns.Rating {
 			// Check if the e.target is a label with the Enum.RatingCssClass.RatingInput class
 			const isInput = Helper.Dom.Styles.ContainsClass(currentTarget, Enum.CssClass.RatingInput);
 			if (isInput) {
+				const lastchosen = this.selfElement.querySelectorAll(GlobalEnum.HTMLElement.Input)[
```
Comment (BenOsodrac) [inline src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:145]: camelCase and _ prefix, as its private variable

Comment (BenOsodrac) [reply src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:145]: I know the rest of the code here doesn't follow that probably, so ignore this  if you want

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts (hunk 1 above)
Comment (JoaoFerreira-FrontEnd) [reply src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:185]: done

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts (hunk 2 above)
Comment (JoaoFerreira-FrontEnd) [reply src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:189]: done

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts (hunk 3 above)
Comment (JoaoFerreira-FrontEnd) [reply src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:354]: done

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts (hunk 4 above)
Comment (JoaoFerreira-FrontEnd) [reply src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:384]: done

File: src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts (hunk 5 above)
Comment (JoaoFerreira-FrontEnd) [reply src/scripts/OSFramework/OSUI/Pattern/Rating/Rating.ts:145]: done

**Review by bmarcelino-fe — APPROVED** [review]

**Pull request merged**